raster/r.proj/testsuite/test_r_proj.py

```diff
@@ -0,0 +1,302 @@
+from grass.gunittest.case import TestCase
+from grass.gunittest.main import test
+
+
+class TestProj(TestCase):
+    """
+
+    Used dataset: nc_spm_08_grass7
+
+    Test cases:
+    test_same_resolution: Test output with region using the same resolution.
+    test_different_resolution: Test output with region using a different resolution.
+    """
+
+    location = "nc_spm_08_grass7"
```

Comment:
Currently, r.proj requires a location parameter, and although the workflow uses nc_spm dataset, they have different names in different context. Changing the hardcoded location to "nc_spm_full_v2alpha2" would probably solve the build issues. But is there a better way to handle this?

Reply:
This is where the testing framework design doesn't work that well. Another location with data is needed, but this test uses the NC SPM as another, so I don't know, perhaps it assumes it is not in NC SPM which is not true when the tests are automatically executed.

Maybe it is good enough test if the same location is used as both current and the other (I'm not sure how r.proj behaves in that case, e.g., if it tries to skip some the steps and just copy). If this is the case, getting the current location name would solve the naming issue.

The more advanced solution is to create a new location (e.g. NC SPF - EPSG:2264) and make it the current one. That's likely better done in a separate environment (copy of os.environ). See, e.g., code in `r.import` or `v.import` or the [create_environment function doc](https://grass.osgeo.org/grass79/manuals/libpython/script.html#script.core.create_environment). You will need to leave out the `use_temp_region` part as it currently does not support passing of the `env` parameter.

Reply:
Thanks! The pointer to the doc for creating new location is really useful. The current behavior of r.proj should perform the same computation (so it's not a direct copy) but show a warning that we are using the same location. Since for now it's more important for me to check the integrity of computation after applying parallelization, I will leave doing the advanced step for the future.